crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -291,25 +278,32 @@ pub(crate) struct TypeInference<'db> {
     /// The diagnostics for this region.
     diagnostics: TypeCheckDiagnostics,
 
-    /// The scope belong to this region.
+    /// The scope this region is part of.
     scope: ScopeId<'db>,
+
+    /// The fallback type for missing expressions/bindings/declarations.
+    missing_ty: Option<Type<'db>>,
 }
 
 impl<'db> TypeInference<'db> {
-    pub(crate) fn empty(scope: ScopeId<'db>) -> Self {
+    pub(crate) fn empty(scope: ScopeId<'db>, missing_ty: Option<Type<'db>>) -> Self {
```

Comment:
NIT: I've a small preference to create a dedicated constructor method, e.g. `TypeInference::cycle_fallback(scope: ScopeId, type: Type)` that enforces a more strict API and may also allow us to have a stricter visibility (private?)